EmeraldComponents/Components/Label/EmeraldLabel.swift

```diff
@@ -97,197 +85,164 @@ public class EmeraldLabel: UIView {
     
     // MARK: - Public configuration
     
-    /// Sets the icon that will be displayed to the left of label.
-    ///
-    /// - Parameter name: The name of an image inside emerald project assets. You can't choose any image. Only emerald images
-    public func setIcon(name: String) {
-        self.auxIconLabel = name
-    }
-    
-    /// Sets the text to be displayed by the EmeraldLabel.
-    ///
-    /// - Parameter Text: The text label name.
-    public func setLabel(text: String) {
-        self.auxTextLabel = text.uppercased()
-    }
-    
     /// Sets the status and type
     ///
     /// - Parameters:
-    ///   - state: Label state to be displayed
+    ///   - text: Label text to be displayed
+    ///   - color: Label color to be displayed
     ///   - type: Label type to be displayed
-    public func configure(state: EmeraldLabelState? = nil, type: EmeraldLabelType? = nil) {
-        self.state = state ?? self.state
+    public func configureFor(text: String,
+                             color: UIColor? = nil,
+                             type: EmeraldLabelType? = nil,
+                             and icon: UIImage? = nil) {
```

Comment:
```suggestion
                             andIcon icon: UIImage? = nil) {
```
Maybe like this it gets more human readable?